src/Components/Form/Form.js

```diff
@@ -19,24 +19,35 @@ export class Form extends Component {
     this.setState({prompts: await getAllQuestions()})
   }
 
-  updateCurrentAnswers = (event) => {
+  updateCurrentAnswers = (event, questionData) => {
     event.preventDefault();
-    if (this.state.currentAnswers.includes(event.target.textContent)){
+    if (this.state.currentAnswers.includes(event.target.textContent)) {
       const answersPerQuestion = [...this.state.currentAnswers]
       const index = this.state.currentAnswers.indexOf(event.target.textContent)
       answersPerQuestion.splice(index, 1)
-      this.setState({currentAnswers: answersPerQuestion })
-      return 
+      return this.setState({currentAnswers: answersPerQuestion})
+    }
+    if (+questionData.allowedResponses && this.state.currentAnswers.find(answer => questionData.choices.includes(answer))) {
+      let choiceToSwap = this.state.currentAnswers.find(answer => questionData.choices.includes(answer))
+      let indexToReplace = this.state.currentAnswers.indexOf(choiceToSwap)
+      let answerToReplace = this.state.currentAnswers
+      answerToReplace.splice(indexToReplace, 1, event.target.textContent)
+      if (answerToReplace) {
+        return this.setState({currentAnswers: answerToReplace})
+      }
+    } else {
+      this.setState({currentAnswers: [...this.state.currentAnswers, event.target.textContent]})
```

Comment:
I have noticed some weird functionality on clicking where the current answers doesn't always get updated. I hope this function isn't the reason for that but think its definitely worth considering